feature: implement cli module

Yihen-Liu opened the pull request:
Signed-off-by: liuyihen@onchain.com

laizy commented on http/websocket/session/session.go:
```diff
@@ -87,3 +90,22 @@ func (self *Session) SessionTimeoverCheck() bool {
+	}
+	for _, v := range rmtxs {
+		delete(self.TxHashMap, v)
+	}
```
key in map can be deleted in iteration

laizy commented on http/websocket/session/session.go:
```diff
@@ -25,13 +25,15 @@ import (
 	mConnection *websocket.Conn
 	nLastActive int64
 	sessionId   string
+	TxHashMap    map[string]int64
```
use array should be more performant

laizy commented on http/websocket/session/session.go:
```diff
@@ -87,3 +95,22 @@ func (self *Session) SessionTimeoverCheck() bool {
+	for k, v := range self.TxHashArr {
+		if (time.Now().Unix() - v.StartTime) < int64(cfg.DEFAULT_GEN_BLOCK_TIME*10) {
+			remove := self.TxHashArr[0:k]
+			self.TxHashArr = self.TxHashArr[k:]
```
need fix

Yihen-Liu committed:
Head line: add cli implement first version.
  add cmd package for cli module.
  restruct main.go file.

  Signed-off-by:liuyihen@onchain.com

Yihen-Liu committed:
<style>: Delete and correct invalid statement
  1. delete unnecessary print in main.go
  2. delete invalid return statement in NewClient()
  3. correct sub-command describtion

  Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
fix: implement the communication by rpc not local-call

Signed-off-by:liuyihen@onchain.com

Yihen-Liu committed:
Refactor: refactor communication from local call to rpc call finally

Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
docs: add some parameter description about cli

Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
refactor: remove some unnecessary code in this version.

  Signed-off-by:liuyihen@onchain.com

Yihen-Liu committed:
style: add package will be used in glide.yaml

  Signed-off-by: liuyihen@onchain.com

Arbio5zt reviewed (changes requested):
1. cli seems no need to use actor.
2. could you redesign the `cmd helps` display formates. such as display in levels.

Yihen-Liu committed:
refactor: modify help info show ,when use ./ontology --help[-h]

  Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
refactor: remove actor code, which is unnecessary.
  Signed-off-by: liuyihen@onchain.com

laizy commented on cmd/cmd.go:
```diff
@@ -0,0 +1,244 @@
+func rpcAddress() string {
+	//return "http://139.219.108.204:20336"
+	index := rand.Intn(len(config.Parameters.SeedList))
+	return "http://" + (strings.Split(config.Parameters.SeedList[index], ":"))[0] + ":" + strconv.Itoa(config.Parameters.HttpJsonPort)
```
need use localhost ip , instead of connecting a remote node.

Yihen-Liu replied:
Thx. Have fix

Yihen-Liu committed:
fix: modified rpc address is localhost
  Signed-off-by: liuyihen@onchain.com

Commit:
Optimizes help tips for block info command.

Optimizes help tips for block info command.

Signed-off-by: XiaoJie Guo <guoxiaojie@onchain.com>

Commit:
Optimizes help tips for transaction info command.

Optimizes help tips for transaction info command.

Signed-off-by: XiaoJie Guo <guoxiaojie@onchain.com>

Commit:
Optimizes help tips for wallet command.

Optimizes help tips for wallet command.

Signed-off-by: XiaoJie Guo <guoxiaojie@onchain.com>

Commit:
Optimizes help tips for set command.

Optimizes help tips for set command.

Signed-off-by: XiaoJie Guo <guoxiaojie@onchain.com>

Yihen-Liu committed:
refactor: modify cmd description by xiaojie
  Signed-off-by: liuyihen@onchain.com

Commit:
Fix issue : wrong passswd result cli panic.

1.Fix issue : wrong passswd result cli panic.
2.Some other minimal changes.

Signed-off-by: XiaoJie Guo <guoxiaojie@onchain.com>

Yihen-Liu committed:
fix: add error log when deploy contract; getversion rpc return is string

  Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
refactor: code style & help information show &show style

  Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
fix: support different wallet balance show
  Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
refactor: when transfer asset, use base58 address
  Signed-off-by: liuyihen@onchain.com

Arbio5zt commented on cmd/contract_cmd.go:
```diff
@@ -0,0 +1,188 @@
+func getVmType(vmType uint) types.VmType {
+	switch vmType {
+	case 0:
+		return types.Native
```
there has no way to deploy native contract, could please remove it ,thanks.

Yihen-Liu replied:
Thx. have removed.

Yihen-Liu committed:
<refactor>: add error handle statement in invalid process flow; correct
 help infomation
  Signed-off-by: liuyihen@onchain.com

laizy commented on cmd/utils/flags.go:
```diff
@@ -0,0 +1,171 @@
+	//contract deploy
+	ContractVmTypeFlag = cli.UintFlag{
+		Name:  "type",
+		Usage: "contract type ,value: NEOVM | NATIVE | SWAM",
```
fix this

Yihen-Liu replied:
Thx. have fix

Yihen-Liu committed:
feat: add transfer asset status query in asset-cli

  Signed-off-by: liuyihen@onchain.com

Yihen-Liu committed:
fix: remove Native smart contract type
  Signed-off-by: liuyihen@onchain.com
